Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrType.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+package org.jetbrains.kotlin.native.interop.gen
+
+sealed class StubType {
+    abstract val nullable: Boolean
+}
+
+/**
+ * Wrapper over [Classifier].
+ */
+class ClassifierStubType(
+        val classifier: Classifier,
+        val typeArguments: List<TypeArgument> = emptyList(),
+        override val nullable: Boolean = false
+) : StubType() {
+    fun nested(name: String): ClassifierStubType =
+            ClassifierStubType(classifier.nested(name))
+}
+
+/**
+ * @return type from kotlinx.cinterop package
+ */
+fun getRuntimeType(name: String, nullable: Boolean = false): ClassifierStubType =
+        ClassifierStubType(Classifier.topLevel("kotlinx.cinterop", name), nullable = nullable)
+
+/**
+ * Functional type from kotlin package: ([parameterTypes]) -> [returnType]
+ */
+class FunctionalType(
+      val parameterTypes: List<StubType>,
+      val returnType: StubType,
+      override val nullable: Boolean = false
+) : StubType() {
+    val classifier: Classifier =
+            Classifier.topLevel("kotlin", "Function${parameterTypes.size}")
+}
+
+class TypeParameterType(
+        val name: String,
+        override val nullable: Boolean
+) : StubType()
+
+/**
+ * Represents typealias.
+ * typealias [abbreviation] = [expandedType].
+ * It's required for typealiases that are declared in kotlinx.cinterop
+ * (e.g. ByteVar, IntVar, etc.)
+ * because metadata expects expanded type, not abbreviation.
+ */
+class AbbreviationStubType(
```

Comment:
It probably should also be applied for all other types that use `typealias` as classifier, for example `typedef`-based and non-strict-`enum`-based ones.